### PR TITLE
Row operations

### LINKED ISSUE #135: [B] Create functionality to detect rows by conditions
Scenario: User specifies that the stored procedure should edit a record and change it to a certain value

This ticket will be purely to create the helper functions necessary for displaying the final results. Given parameters, return a list of matching rows.

Given a table, detect if a record does exist where attribute 

- Exactly matches value
- Contains value
- Does not contain value
- Has ID which exists in test data
- Has ID given by user

### PR DESCRIPTION
**Pull request by adityavidyadharan**
# Description

Functions to handle row based conditions. Also includes changes to entity structure.

## Link to the issue you are closing

Fixes #135 

## Screenshot or other testing that you have completed

```ts
public async test() {
    await AppDataSource.initialize();

    const p = new PgClient({
      nickname: 'asdf',
      type: DBProvider.PostgreSQL,
      id: 1,
      defaultDatabase: 'React',
      address: 'localhost',
      password: 'asdf',
      username: 'kpmg',
      port: 5432,
    });
    console.log('Exact');
    console.log(await p.checkExact('accounts', 'name', 'Bob'));
    console.log('ID');
    console.log(await p.checkID('accounts2', 'accounts'));
    console.log('Number, LT');
    console.log(
      await p.checkNumber('accounts', 'balance', 9999, RowNumberOperations.LT)
    );
    console.log('Number, GT');
    console.log(
      await p.checkNumber('accounts', 'balance', 9999, RowNumberOperations.GT)
    );
    console.log('Boolean');
    console.log(
      await p.checkBoolean('accounts2', 'valid', RowBooleanOperations.TRUE)
    );

    const rep = AppDataSource.getRepository(RuleEntity);

    const t = new TableTestEntity();
    t.operation = TableGenericOperations.COUNT;
    t.name = 'test';
    t.table = 'accounts';
    t.value = '1';
    t.result = false;
    t.format = OutputFormat.JSON;
    t.output = '';
    t.level = UnitTestOperations.TableGenericOperations;

    const s = new RuleEntity();
    s.name = 'test';
    s.database = 'React';
    s.ruleId = 1;
    s.unitTests = [t];
    s.testData = 'accounts';
    rep.save(s);

    console.log(plainToInstance(RuleEntity, instanceToPlain(s)));
    console.log(
      plainToInstance(RuleEntity, {
        unitTests: [{ level: UnitTestOperations.TableGenericOperations }],
      })
    );
    console.log(
      plainToInstance(RuleEntity, {
        unitTests: [{ level: UnitTestOperations.RowStringOperations }],
      })
    );
  }
```

# Checklist:

- [x] ESLint passes
- [x] Prettier passes
- [x] Included comments where necessary
- [x] Updated README if needed
- [x] Verified that code runs and generates no errors/warnings

### CODE REVIEW

File: src/db/clients/PgClient.ts [hunk 1]
```diff
@@ -110,4 +111,84 @@ export default class PgClient implements ServerInterface {
...
+    column: string,
+    value: RowBooleanOperations
+  ): Promise<unknown[]> {
+    const client = await this.pool.connect();
+    const result = await client.query(
+      `select * from ${table} where ${column} = ${value}`
+    );
+    client.release();
+    return result.rows;
+  }
```
Comment (insaan21) [inline src/db/clients/PgClient.ts:193]: What is the difference between checkBoolean and checkExact? the code is the same

File: src/db/clients/PgClient.ts [hunk 2]
```diff
@@ -110,4 +111,84 @@ export default class PgClient implements ServerInterface {
...
+  ): Promise<unknown[]> {
+    const client = await this.pool.connect();
+    const result = await client.query(
+      `select * from ${table} where ${column} like '${value}'`
+    );
+    client.release();
+    return result.rows;
+  }
+
+  public async checkID(data: string, table: string): Promise<unknown[]> {
```
Comment (insaan21) [inline src/db/clients/PgClient.ts:155]: What does checkID do?

File: src/db/models/UnitTestModels.ts [hunk 3]
```diff
@@ -1,22 +1,39 @@
...
-  count?: number;
-}
+  table: string; // table to run test on
+
+  column?: string;
 
-export interface RowTestType extends UnitTestType {
-  operation: RowOperations;
-  column: string;
   value?: string;
```
Comment (insaan21) [inline src/db/models/UnitTestModels.ts:23]: is this supposed to represent count for the table test operations, if so I'll add a comment

File: src/db/clients/PgClient.ts (hunk 1 above)
Comment (adityavidyadharan) [reply src/db/clients/PgClient.ts:193]: You're right, I'll combine the two

File: src/db/clients/PgClient.ts (hunk 2 above)
Comment (adityavidyadharan) [reply src/db/clients/PgClient.ts:155]: check if the IDs in the input table (data) exist in the target table (table)

File: src/db/models/UnitTestModels.ts (hunk 3 above)
Comment (insaan21) [reply src/db/models/UnitTestModels.ts:23]: Nvm resolved

File: src/db/clients/PgClient.ts (hunk 2 above)
Comment (insaan21) [reply src/db/clients/PgClient.ts:155]: Got it

**Review by insaan21 — APPROVED** [review]
lgtm!

**Pull request merged**